src/webapi/node.rs

```diff
@@ -362,6 +362,58 @@ pub struct Node( Reference );
 impl IEventTarget for Node {}
 impl INode for Node {}
 
+impl Node {
+    /// Attempt to create the `Node` from raw html.
+    ///
+    /// # Panics
+    ///
+    /// This method panics if the resulting number of nodes != 1. To ensure this
+    /// it is recommended to at _least_ have programatic control of your root
+    /// HTML node.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let node = Node::from_html("<div>Some text, horray!</div>").unwrap();
+    /// ```
+    pub fn from_html(html: &str) -> Result<Node, SyntaxError> {
+        let result = js! {
+            var span = document.createElement("span");
+            try {
+                span.innerHTML = @{html};
+            } catch( error ) {
```

Comment:
You can probably use the `js_try!` macro here, you know. It should be simpler. (:

Reply:
`js_try!` doesn't return a value currently and I couldn't get it to without a lot of fiddling (and breaking tests... and regressing it's performance), so I opted for this route. It could be replaced in the future if it _did_ return a value.

Reply:
But it does return a value? (: (See the `js_try` test in `macros.rs`)

Reply:
ah crap, I only looked at the `no_return` pattern :laughing: